src/selectors/entities/channel_categories.ts

```diff
@@ -189,6 +189,76 @@ export function makeFilterAutoclosedDMs(getAutocloseCutoff = getDefaultAutoclose
     );
 }
 
+export function makeFilterAutoclosedDMs(): (state: GlobalState, channels: Channel[], categoryType: string) => Channel[] {
+    return createSelector(
+        (state: GlobalState, channels: Channel[]) => channels,
+        (state: GlobalState, channels: Channel[], categoryType: string) => categoryType,
+        getMyPreferences,
+        shouldAutocloseDMs,
+        getCurrentChannelId,
+        (state: GlobalState) => state.entities.users.profiles,
+        getCurrentUserId,
+        getMyChannelMemberships,
+        (channels, categoryType, myPreferences, autocloseDMs, currentChannelId, profiles, currentUserId, myMembers) => {
+            if (categoryType !== CategoryTypes.DIRECT_MESSAGES) {
+                // Only autoclose DMs that haven't been assigned to a category
+                return channels;
+            }
+
+            let unreadCount = 0;
+
+            const filtered = channels.filter((channel) => {
+                if (channel.type !== General.DM_CHANNEL && channel.type !== General.GM_CHANNEL) {
+                    return true;
+                }
+
+                if (isUnreadChannel(myMembers, channel)) {
+                    unreadCount++;
+
+                    // Unread DMs/GMs are always visible
+                    return true;
+                }
+
+                if (currentChannelId === channel.id) {
+                    // The current channel is always visible
+                    return true;
+                }
+
+                // openTime is the time the channel was last opened (like from the More DMs list) after having been closed
+                const openTimePref = myPreferences[getPreferenceKey(Preferences.CATEGORY_CHANNEL_OPEN_TIME, channel.id)];
```

Comment:
Instead of looking at this, let's compare the deactivated time against the channel member's `last_viewed_at` since that's a lot more widely used. The OPEN_TIME preference is something I didn't know existed until I started digging into this, and I'm pretty sure it has issues where it's not set right. I'd like to remove it eventually.

Reply:
Oh yes I remember you did mention that, I forgot this part of the conversation SORRY!